### PR TITLE
Fix paths in .coverage file in Python 3

### PR DESCRIPTION
**Pull request by astrofrog**
Fixes paths in .coverage file in Python 3 like we do in Python 2, otherwise coverage reporting to e.g. coveralls does not work in Python 3. This will not work in Python 3 for packages that still use 2to3, but this never worked anyway since the temporary files got deleted before.

This should fix https://github.com/astropy/package-template/issues/151

### CODE REVIEW

**Comment by astrofrog** [comment]
I'm also enabling coverage testing on Python 3 in the ``.travis.yml`` file to make sure that things work correctly (this is still back-portable to 1.0.x)

**Comment by astrofrog** [comment]
@astropy/astropy-core-maintainers - this is now ready for review. However, the Travis build on which I enabled Python 3 coverage testing goes from 8 minutes to 24 minutes. I've opened an issue to remind us to investigate why running with coverage testing is so much slower on Python 3 (https://github.com/astropy/astropy/issues/4826).

If people are opposed to the increase in testing time, I can disable the Python 3 coverage testing again? (at least we know it works now)

**Comment by mhvk** [comment]
@astrofrog - great that it now works well on python3! I think we should definitely opt for either python2 or python3 for coverage, not do both. My sense would be to stick to python2 until we figure out why python3 takes so much longer, i.e., I would suggest not to make the change to `.travis.yml`.

**Comment by astrofrog** [comment]
@mhvk - sounds good - I just wanted to make the change as part of this PR to make sure it works, but happy to remove it.

**Comment by pllim** [comment]
Hmm. Does it also take 24 minutes locally? If so, maybe profiling can pinpoint where it is so slow.

File: .travis.yml [hunk 1]
```diff
@@ -64,18 +64,19 @@ matrix:
...
+        # build (with latest numpy). We also note the code coverage.
         - os: linux
           env: PYTHON_VERSION=2.7 SETUP_CMD='test --coverage'
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
                PIP_DEPENDENCIES='cpp-coveralls objgraph'
                LC_CTYPE=C.ascii LC_ALL=C
                CFLAGS='-ftest-coverage -fprofile-arcs -fno-inline-functions -O0'
         - os: linux
-          env: PYTHON_VERSION=3.4 SETUP_CMD='test'
+          env: PYTHON_VERSION=3.4 SETUP_CMD='test --coverage'
```
Comment (pllim) [inline .travis.yml]: Why not 3.5?

Comment (astrofrog) [inline .travis.yml]: I just left it as-is for now, I think that should be changed in a separate PR.

**Comment by astrofrog** [comment]
@pllim - yes, it also takes a while locally, for me at least. I'm not sure on the best way to profile the tests because it seems to be an issue that occurs only when the coverage testing is working, so it's not just about profiling individual astropy functions or operations. I don't really have time to look into it now, but we should discuss it in https://github.com/astropy/astropy/issues/4826

File: .travis.yml (hunk 1 above)
Comment (pllim) [inline .travis.yml]: I am also wondering if it is also slow in Python 3.5 or just 3.4.

**Comment by astrofrog** [comment]
I've now removed the commit that changed the ``.travis.yml`` file, so this can be merged once people are happy with it.

File: .travis.yml (hunk 1 above)
Comment (astrofrog) [inline .travis.yml]: Ah I see what you mean - it's slow on Python 3.5 too (I see it locally)

**Comment by mhvk** [comment]
@astrofrog - OK, let's move over to trying to figure out why coverage is so much slower on python3. Merging this aspect so that testing can be done on master!
